src/daemon/rpmostreed-transaction-types.c

```diff
@@ -1054,6 +1057,12 @@ initramfs_state_transaction_execute (RpmostreedTransaction *transaction,
   rpmostree_origin_set_regenerate_initramfs (origin, self->regenerate, self->args);
   rpmostree_sysroot_upgrader_set_origin (upgrader, origin);
 
+  RpmOstreeSysrootUpgraderLayeringType layering_type;
+  gboolean layering_changed = FALSE;
+  if (!rpmostree_sysroot_upgrader_prep_layering (upgrader, &layering_type, &layering_changed,
```

Comment:
Makes sense, though I kept the assert since eh, assertions are cheap.  Fixup :arrow_down: